Sources/APIServer/Volumes/VolumesService.swift

```diff
@@ -0,0 +1,220 @@
+        }
+
+        // Check if volume already exists by trying to list and finding it
+        let existingVolumes = try await store.list()
+        if existingVolumes.contains(where: { $0.name == name }) {
+            throw VolumeError.volumeAlreadyExists(name)
+        }
+
+        try createVolumeDirectory(for: name)
+
+        // Parse size from driver options (default 512GB)
+        let sizeInBytes: UInt64
+        if let sizeString = driverOpts["size"] {
+            sizeInBytes = try parseSize(sizeString)
+        } else {
+            sizeInBytes = VolumeStorage.defaultVolumeSizeBytes
+        }
+
+        try createVolumeImage(for: name, sizeInBytes: sizeInBytes)
+
+        let volume = Volume(
+            name: name,
+            driver: driver,
+            format: "ext4",
+            source: blockPath(for: name),
+            labels: labels,
+            options: driverOpts
+        )
+
+        try await store.create(volume)
+
+        log.info("Created volume", metadata: ["name": "\(name)", "driver": "\(driver)"])
+        return volume
+    }
+
+    public func delete(name: String) async throws {
+        try await lock.withLock { _ in
+            try await self._delete(name: name)
+        }
+    }
+
+    private func _delete(name: String) async throws {
+        guard VolumeStorage.isValidVolumeName(name) else {
+            throw VolumeError.invalidVolumeName("Invalid volume name '\(name)': must match \(VolumeStorage.volumeNamePattern)")
+        }
+
+        // Check if volume exists by trying to list and finding it
+        let existingVolumes = try await store.list()
+        guard existingVolumes.contains(where: { $0.name == name }) else {
+            throw VolumeError.volumeNotFound(name)
+        }
+
+        // Check if volume is in use by any container
+        if let containersService = self.containersService, await containersService.isVolumeInUse(name) {
+            throw VolumeError.volumeInUse(name)
+        }
+
+        try await store.delete(name)
+        try removeVolumeDirectory(for: name)
+
+        log.info("Deleted volume", metadata: ["name": "\(name)"])
+    }
+
+    public func list() async throws -> [Volume] {
+        try await store.list()
+    }
+
+    public func inspect(_ name: String) async throws -> Volume {
+        try await lock.withLock { _ in
+            try await self._inspect(name)
+        }
+    }
+
+    private func _inspect(_ name: String) async throws -> Volume {
+        guard VolumeStorage.isValidVolumeName(name) else {
+            throw VolumeError.invalidVolumeName("Invalid volume name '\(name)': must match \(VolumeStorage.volumeNamePattern)")
+        }
+
+        let volumes = try await store.list()
+        guard let volume = volumes.first(where: { $0.name == name }) else {
+            throw VolumeError.volumeNotFound(name)
+        }
+
+        // Return volume without usage data since we removed refcounting
```

Comment:
Oh crud, one final thing. I'd remove/change this comment. No one reading this after should have to know the lore of this. Although possibly add something like:
```
// NOTE: It may be useful to return some sort of refcount showing
// what containers are using this volume eventually.
```